components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/dao/JDBCSubscriberDAOImpl.java

```diff
@@ -266,38 +312,39 @@ private void insertSubscriber(String subscriberId, PublisherDataHolder holder, P
         }
     }
 
-    private void updateSubscriber(String subscriberId, PublisherDataHolder holder, PublisherPropertyDTO[] propertyDTOs,
-                                  PublisherDataHolder oldHolder) throws EntitlementException {
-
-        int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
+    /**
+     * DAO method to update a subscriber.
+     *
+     * @param subscriberId        subscriber ID.
+     * @param updatedModuleName   updated module name.
+     * @param updatedPropertyDTOS updated property DTOs.
+     * @param tenantId            tenant ID.
+     * @throws EntitlementException If an error occurs.
+     */
+    private void updateSubscriber(String subscriberId, String updatedModuleName,
+                                  PublisherPropertyDTO[] updatedPropertyDTOS, int tenantId)
+            throws EntitlementException {
 
         Connection connection = IdentityDatabaseUtil.getDBConnection(true);
-        try (NamedPreparedStatement updateSubscriberPrepStmt = new NamedPreparedStatement(connection,
-                UPDATE_SUBSCRIBER_MODULE_SQL);
-             NamedPreparedStatement updateSubscriberPropertiesPrepStmt = new NamedPreparedStatement(connection,
-                     UPDATE_SUBSCRIBER_PROPERTIES_SQL)) {
+        try (NamedPreparedStatement updateSubscriberPropertiesPrepStmt = new NamedPreparedStatement(connection,
+                UPDATE_SUBSCRIBER_PROPERTIES_SQL)) {
 
-            if (!oldHolder.getModuleName().equalsIgnoreCase(holder.getModuleName())) {
-                updateSubscriberPrepStmt.setString(ENTITLEMENT_MODULE_NAME, holder.getModuleName());
+            if (StringUtils.isNotBlank(updatedModuleName)) {
+                NamedPreparedStatement updateSubscriberPrepStmt = new NamedPreparedStatement(connection,
```

Comment:
use try with resources.

Reply:
lets take this out of the update properties logic..

```
if (StringUtils.isNotBlank(updatedModuleName)) {
    // Update module name
}

if (ArrayUtils.isNotEmpty((updatedPropertyDTOS)) {
    // Update properties
}
```

Reply:
What i meant to use try-with-resources for `updateSubscriberPrepStmt`.. Can't we do that?

Reply:
will update